.circleci/orbs/code-infra.yml

```diff
@@ -0,0 +1,214 @@
+version: 2.1
+
+commands:
+  install-deps:
+    description: 'Installs JavaScript dependencies using pnpm'
+    parameters:
+      browsers:
+        type: boolean
+        default: false
+        description: 'Set to true if you intend to install any browser (for example with playwright).'
+      package-overrides:
+        description: 'A space separated list of package@version to override during installation'
+        type: string
+        default: ''
+      ignore-workspace:
+        description: 'Set to true to ignore the pnpm workspace (useful for single-package repositories)'
+        type: boolean
+        default: false
+    steps:
+      - run:
+          name: Set npm registry public signing keys
+          command: |
+            echo "export COREPACK_INTEGRITY_KEYS='$(curl https://registry.npmjs.org/-/npm/v1/keys | jq -c '{npm: .keys}')'" >> $BASH_ENV
+      - when:
+          condition: << parameters.browsers >>
+          steps:
+            - run:
+                name: Install pnpm package manager
+                command: corepack enable
+      - when:
+          condition:
+            not: << parameters.browsers >>
+          steps:
+            - run:
+                name: Install pnpm package manager
+                # See https://stackoverflow.com/a/73411601
+                command: corepack enable --install-directory ~/bin
+      - run:
+          name: View install environment
+          command: |
+            node --version
+            pnpm --version
+      - when:
+          condition: << parameters.ignore-workspace >>
+          steps:
+            - run:
+                name: Initialize single-package repository
+                command: pnpm install --ignore-workspace
+      - when:
+          condition:
+            not: << parameters.ignore-workspace >>
+          steps:
+            - run:
+                name: Install js dependencies
+                command: |
+                  packages="<< parameters.package-overrides >>"
```

Comment:
If we have to do so much processing to parse `parameters.package-overrides`, perhaps we should just do it in the CLI? Keep complexity of the orb low.